lib/samples/download_preplanned_map_area/download_preplanned_map_area_sample.dart

```diff
@@ -34,130 +33,280 @@ class DownloadPreplannedMapAreaSample extends StatefulWidget {
 
 class _DownloadPreplannedMapAreaSampleState
     extends State<DownloadPreplannedMapAreaSample> with SampleStateSupport {
+  // Create a controller for the map view.
   final _mapViewController = ArcGISMapView.createController();
-  var _preplannedAreas = <PreplannedMapArea>[];
-  var _selectedPreplannedAreaIndex = -1;
-  var _isLoading = false;
-  OfflineMapTask? _offlineMapTask;
-  final _sourceMap = ArcGISMap.withUri(Uri.parse(
-      'https://arcgisruntime.maps.arcgis.com/home/item.html?id=acc027394bc84c2fb04d1ed317aac674'))!;
+  // Prepare an offline map task and map for the online web map.
+  late OfflineMapTask _offlineMapTask;
+  late ArcGISMap _webMap;
+  // The location to save offline maps to.
+  Directory? _downloadDirectory;
+  // Create a Map to track preplanned map areas and their associated download jobs.
+  final _preplannedMapAreas =
+      <PreplannedMapArea, DownloadPreplannedOfflineMapJob?>{};
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+  // A flag for when the map selection UI should be displayed.
+  var _mapSelectionVisible = false;
+
+  @override
+  void dispose() {
+    // Delete downloaded offline maps on exit.
+    _downloadDirectory?.deleteSync(recursive: true);
+    super.dispose();
+  }
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       body: SafeArea(
-        child: Column(
-          crossAxisAlignment: CrossAxisAlignment.center,
+        child: Stack(
           children: [
-            Expanded(
-              child: ArcGISMapView(
-                controllerProvider: () =>
-                    _mapViewController..arcGISMap = _sourceMap,
-                onMapViewReady: onMapViewReady,
+            Column(
+              children: [
+                Expanded(
+                  // Add a map view to the widget tree and set a controller.
+                  child: ArcGISMapView(
+                    controllerProvider: () => _mapViewController,
+                    onMapViewReady: onMapViewReady,
+                  ),
+                ),
+                Row(
```

Comment:
Can this `Row` be replaced by a `Center`? There is just the button.

Reply:
Yeah it could, but I think we prefer Row since it's consistent across the samples.